Pcc62

Gealber opened the pull request:
ATTENTION: before clicking "Create Pull Request" please submit some meta data, thanks!

Difficulty level (1-10): [8]
Estimated time spent (hours): [4]
Completed (yes/no): [yes]
I stretched my coding skills (if yes what did you learn?): [I learned to use threading and matplotlib]
Other feedback (what can we improve?): []

clamytoe commented:
Whoa, you’re committing the whole repo my man. Close it and try again.

bbelderbos commented:
I changed base branch to "community", should be fine now

Gealber commented:
Sorry I'm new with git and programing in general. Thanks 


De: "Martin Uribe" <notifications@github.com> 
Para: "pybites/challenges" <challenges@noreply.github.com> 
CC: "Gealber Morales" <gealberm@uci.cu>, "Author" <author@noreply.github.com> 
Enviados: Viernes, 19 de Julio 2019 18:32:39 
Asunto: Re: [pybites/challenges] Pcc62 (#547) 



Whoa, you’re committing the whole repo my man. Close it and try again. 

— 
You are receiving this because you authored the thread. 
Reply to this email directly, [ https://github.com/pybites/challenges/pull/547?email_source=notifications&email_token=ALRB6E4BRAR6A7QARL42H7DQAI6IPA5CNFSM4IFMW6X2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2M5ZRI#issuecomment-513400005 | view it on GitHub ] , or [ https://github.com/notifications/unsubscribe-auth/ALRB6E5S5TV27RQ5YCSL6E3QAI6IPANCNFSM4IFMW6XQ | mute the thread ] . 

1519-2019: Aniversario 500 de la Villa de San Cristóbal de La Habana
Por La Habana, lo más grande. #Habana500 #UCIxHabana500

pmayd commented on 62/Gealber/scrapper.py:
```diff
@@ -0,0 +1,140 @@
+        d = gender.Detector()
+        for year in speakers_in_year_es:
+            for i, name in enumerate(speakers_in_year_es[year]):
+                speakers_in_year_es[year][i] = 1 if d.get_gender(name.split()[0]) == 'female' else 0
```
This is not very Pythonic and could be rewritten using list comprehensions, at least for the inner loop. No need for the "i"

Gealber commented:
Thank you very much for the advice, and you should continue doing it, it benefits me a lot especially for me that I am a mathematician and not a programmer. So I have a very bad base in terms of knowledge in Computer Science(and English). 
Really really helpful. 


De: "Michael Aydinbas" <notifications@github.com> 
Para: "pybites" <challenges@noreply.github.com> 
CC: "Gealber Morales" <gealberm@uci.cu>, "Author" <author@noreply.github.com> 
Enviados: Domingo, 13 de Octubre 2019 3:03:02 
Asunto: Re: [pybites/challenges] Pcc62 (#547) 



@pmayd approved this pull request. 

Very nice job and good use of libraries 


In [ https://github.com/pybites/challenges/pull/547#discussion_r334265276 | 62/Gealber/scrapper.py ] : 
> +            print(f"The following error had raise while trying to connect to {url}\n")
+            print(e) 


you can combine several outputs with one print function if you want print("a","b", sep="\n") . As you can see yu can even specify the separator 


In [ https://github.com/pybites/challenges/pull/547#discussion_r334265327 | 62/Gealber/scrapper.py ] : 
> +        the speakers in a list.
+        :param html:
+        :param year:
+        :return speakers: List of speakers' full name.
+        """
+        soup = BeautifulSoup(html, "lxml")
+        speakers = []
+        years_common = [2015, 2016, 2017]
+        if year == 2018:
+            items = soup.select("div.schedule__talk--speaker > div > p")
+            for i, link in enumerate(items):
+                s = str(link)
+                start_content = s.find('<p>')
+                end_content = s.find('</p>', start_content + 1)
+                speaker = str(s[start_content + 3: end_content]).replace(' y', ',').split(', ')
+                speakers = speakers + speaker 


this is no good idea in Python as "+" creates a new list which is much more memory consumable than using append , which only prolongs the existing list but does not create a new one. So you should use speakers.append(speaker) 


In [ https://github.com/pybites/challenges/pull/547#discussion_r334265437 | 62/Gealber/scrapper.py ] : 
> +            items = soup.select("div.slot-inner > p > strong")
+            for i, link in enumerate(items):
+                s = str(link)
+                start_content = s.find('<strong>')
+                end_content = s.find('</strong>', start_content + 1)
+                speaker = str(s[start_content + 8: end_content]).replace(' y', ',').split(', ')
+                speakers = speakers + speaker 


this is too much repetition. The only thing that changes here is the item you select and the offset for start_content. But this offset can be calculated by len(html-tag). So you should really refactor this code and work for example with a dictionary that has as key the year and as value a tuple with (tag, start_tag, end_tag) and then you can easily use this information in one gothrough and not repeat yourself. When you have to repeat yourself more than once its time to refactor 

— 
You are receiving this because you authored the thread. 
Reply to this email directly, [ https://github.com/pybites/challenges/pull/547?email_source=notifications&email_token=ALRB6EZJ46RQJOSAQDKAC7DQOLB2NA5CNFSM4IFMW6X2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCHYW7FI#pullrequestreview-301035413 | view it on GitHub ] , or [ https://github.com/notifications/unsubscribe-auth/ALRB6E7URIZTUG4MD2D3DSLQOLB2NANCNFSM4IFMW6XQ | unsubscribe ] . 

1519-2019: Aniversario 500 de la Villa de San Cristóbal de La Habana
Por La Habana, lo más grande. #Habana500 #UCIxHabana500
